lib/levelup.js

```diff
@@ -53,8 +54,23 @@ function LevelUP (db, options, callback) {
 
   this.options = getOptions(options)
   this._db = db
-  this.db = new DeferredLevelDOWN(db)
+  this.db = db.supports && db.supports.deferredOpen ? db : new DeferredLevelDOWN(db)
   this.open(callback)
+
+  // Create manifest based on db or deferred-leveldown's
+  this.supports = supports(this.db.supports, {
+    status: false,
+    deferredOpen: true,
+    openCallback: true,
+    promises: true,
+    streams: true
+  })
+
+  Object.keys(this.supports.additionalMethods).forEach(function (method) {
+    this[method] = this[method] || function () {
+      return this.db[method].apply(this.db, arguments)
+    }
+  }, this)
```

Comment:
@ralphtheninja Shall I keep this (L69-L73) in, or do it later? Until we have manifests in every module, this `levelup` code won't actually do anything (and can't be tested).

Reply:
No opinion :)